File: app/src/main/java/com/capstone/project_niyakneyak/ui/alarm/service/RescheduleAlarmService.java
package com.capstone.project_niyakneyak.ui.alarm.service;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.lifecycle.LifecycleService;

import android.content.Intent;
import android.os.IBinder;

import com.capstone.project_niyakneyak.data.alarm_resource.AlarmRepository;
import com.capstone.project_niyakneyak.data.alarm_model.Alarm;

public class RescheduleAlarmService extends LifecycleService {
    @Override
    public void onCreate() {
        super.onCreate();
    }

    @Override
    public int onStartCommand(@Nullable Intent intent, int flags, int startId) {
        super.onStartCommand(intent, flags, startId);

        AlarmRepository alarmRepository = new AlarmRepository(getApplication());
        alarmRepository.getAlarmsLiveData().observe(this, alarms -> {
            for (Alarm a : alarms) {
                if (a.isStarted()) {
                    a.scheduleAlarm(getApplicationContext());
                }
            }
        });

        return START_STICKY;
    }

    @Override
    public IBinder onBind(@NonNull Intent intent) {
        super.onBind(intent);
        return null;
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
    }
}

File: app/src/main/java/com/capstone/project_niyakneyak/ui/alarm/service/RescheduleAlarmService.kt
package com.capstone.project_niyakneyak.ui.alarm.service

import android.content.Intent
import android.os.IBinder
import androidx.lifecycle.LifecycleService
import com.capstone.project_niyakneyak.data.alarm_model.Alarm
import com.capstone.project_niyakneyak.data.alarm_resource.AlarmRepository

class RescheduleAlarmService : LifecycleService() {
    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
        super.onStartCommand(intent, flags, startId)
        val alarmRepository = AlarmRepository(application)
        alarmRepository.alarmsLiveData!!.observe(this) { alarms: List<Alarm?>? ->
            for (a in alarms!!) {
                if (a!!.isStarted) {
                    a.scheduleAlarm(applicationContext)
                }
            }
        }
        return START_STICKY
    }

    override fun onBind(intent: Intent): IBinder? {
        super.onBind(intent)
        return null
    }
}